src/AvPortalClient.php

```diff
@@ -28,17 +33,34 @@ class AvPortalClient implements AvPortalClientInterface {
    */
   protected $httpClient;
 
+  /**
+   * The time service.
+   *
+   * @var \Drupal\Component\Datetime\TimeInterface
+   */
+  protected $time;
+
   /**
    * Constructs an AvPortalClient object.
    *
-   * @param \GuzzleHttp\ClientInterface $http_client
+   * @param \GuzzleHttp\ClientInterface $httpClient
    *   The HTTP client.
    * @param \Drupal\Core\Config\ConfigFactoryInterface $configFactory
    *   The config factory.
+   * @param \Drupal\Core\Cache\CacheBackendInterface $cacheBackend
+   *   The cache backend.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   The time service.
+   * @param bool $useCaches
+   *   If the client should use caches for storing and retrieving responses.
    */
-  public function __construct(ClientInterface $http_client, ConfigFactoryInterface $configFactory) {
-    $this->httpClient = $http_client;
+  public function __construct(ClientInterface $httpClient, ConfigFactoryInterface $configFactory, CacheBackendInterface $cacheBackend = NULL, TimeInterface $time = NULL, bool $useCaches = TRUE) {
+    $this->httpClient = $httpClient;
     $this->config = $configFactory->get('media_avportal.settings');
+    $this->cacheBackend = $cacheBackend ?? \Drupal::service('cache.default');
+    $this->time = $time ?? \Drupal::service('datetime.time');
```

Comment:
Do we have cases when this service not available?
It would be nice to left comment in code.

Reply:
This is done for backwards compatibility. If somebody extended the service, it wouldn't be passing the parameters and thus they would be forced to update the code. Drupal does this often to avoid breaking contrib modules (custom code can be easily adapted after a release, but not the contribs you are depending on).